compiler/src/main/java/io/neow3j/compiler/converters/ObjectsConverter.java

```diff
@@ -119,33 +118,49 @@ private AbstractInsnNode handleContractInterfaceCtor(AbstractInsnNode insn, NeoM
         if (hasAnnotations(ownerClassNode, NativeContract.class)) {
             return handleNativeContractHash(neoMethod, ownerClassNode, insn);
         }
-        if (cTorParamLength == 1 && getInternalName(Hash160.class).equals(cTorArgTypes[0].getInternalName())) {
-            // The instructions that lead to the contract hash are already on the stack at this point.
-            return insn;
+        if (cTorParamLength == 1) {
+            if (getInternalName(Hash160.class).equals(cTorArgTypes[0].getInternalName())) {
+                // The instructions that lead to the contract hash are already on the stack at this point.
+                return insn;
+            } else if (getInternalName(String.class).equals(cTorArgTypes[0].getInternalName())) {
+                NeoInstruction lastInstruction = neoMethod.getLastInstruction();
+                handleConstantScriptHash(neoMethod, lastInstruction);
+                return insn;
+            }
         }
         throw new CompilerException(
-                format("A constructor of a ContractInterface is required to take exactly one %s type as parameter.",
-                        getInternalName(Hash160.class)));
+                format("A constructor of a ContractInterface is required to take exactly one %s or %s type as " +
+                        "parameter.", getInternalName(Hash160.class), getInternalName(String.class)));
+    }
+
+    private void handleConstantScriptHash(NeoMethod neoMethod, NeoInstruction lastInstruction) {
+        if (lastInstruction.getOpcode().equals(OpCode.PUSHDATA1)) {
+            io.neow3j.types.Hash160 scriptHash = new io.neow3j.types.Hash160(new String(lastInstruction.getOperand()));
+            neoMethod.replaceLastInstruction(buildPushDataInsn(reverseArray(scriptHash.toArray())));
+        } else {
+            throw new CompilerException("Contract interface classes can only be initialized with Hash160 or " +
```

Comment:
Please write an exception-test for this path.

Reply:
I've added an UT in commit 5a366ff090cefc54e347e3b74732e2095acbff4e.